source/docs/software/advanced-control/pidcontroller.rst

```diff
@@ -0,0 +1,169 @@
+
+Using the ``PIDController`` Class
+---------------------------------
+
+.. note:: The ``PIDController`` class in the ``frc`` namespace is deprecated - C++ teams should use the one in the ``frc2`` namespace, instead.  Likewise, Java teams should use the class in the ``wpilibj.controller`` package.
+
+Constructing a ``PIDController``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. note:: While ``PIDController`` may be used asynchronously, it does *not* provide any thread safety features - ensuring threadsafe operation is left entirely to the user, and thus asynchronous usage is recommended only for advanced teams.
+
+In order to use WPILib's PID control functionality, users must first construct a ``PIDController`` object with the desired gains:
+
+.. tabs::
+
+  .. code-tab:: java
+
+    // Creates a PIDController with gains kP, kI, and kD
+    PIDController pid = new PIDController(kP, kI, kD);
+
+  .. code-tab:: c++
+
+    // Creates a PIDController with gains kP, kI, and kD
+    frc2::PIDController pid(kP, kI, kD);
+
+An optional fourth parameter can be provided to the constructor, specifying the period at which the controller will be run.  The ``PIDController`` object is intended primarily for synchronous use from the main robot loop, and so this value is defaulted to 20ms.
+
+Using the Feedback Loop Output
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. note:: The ``PIDController`` assumes that the ``calculate()`` method is being called regularly at an interval consistent with the configured period.  Failure to do this will result in unintended loop behavior.
+
+.. warning:: Unlike the old ``PIDController``, the new PIDController does not automatically control an output from its own thread - users are required to call ``calculate()`` and use the resulting output in their own code.
+
+Using the constructed ``PIDController`` is simple: simply call the ``calculate()`` method from the robot's main loop (e.g. the robot's ``autonomousPeriodic()`` method):
+
+.. tabs::
+
+  .. code-tab:: java
+
+    // Calculates the output of the PID algorithm based on the sensor reading
+    // and sends it to a motor
+    motor.set(pid.calculate(encoder.getDistance(), setpoint));
+
+  .. code-tab:: c++
+
+    // Calculates the output of the PID algorithm based on the sensor reading
+    // and sends it to a motor
+    motor.Set(pid.Calculate(encoder.GetDistance(), setpoint));
+
+Checking Errors
+^^^^^^^^^^^^^^^
+
+.. note:: ``getPositionError()` and ``getVelocityError()`` are named assuming that the loop is controlling a position - for a loop that is controlling a velocity, these return the velocity error and the acceleration error, respectively.
```

Comment:
Check # of backticks here.
![image](https://user-images.githubusercontent.com/19977330/70481171-c54d5500-1aaf-11ea-816d-8c897ff405b1.png)